### PR TITLE
Add plug-ins and extensions section

### PR DESCRIPTION
**Pull request by sam-roberts-bjss**

### CODE REVIEW

File: app/forms/sections/plug-ins-or-extensions.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  ],
+  "questions": {
+    "plugins-required": {   
+      "mainAdvice": "Are any plug-ins or browser extensions required?*",
+      "additionalAdvice": "Please select below if your Solution requires any plug-ins or browser extentions.",
+      "type": "radiobutton-options",
+      "errorResponse": {
+        "required": "Please select whether or not plug-ins or browser extensions are required"
+      },
+      "options": [
```
Comment (alibombaye) [inline app/forms/sections/plug-ins-or-extensions.json]: Take this guy out. We shouldn't need it anymore

File: app/views/preview/components/preview-section-table-row.njk [hunk 2]
```diff
@@ -1,11 +1,11 @@
 {% macro previewSectionTableRow(questionId, questionTitle, innerComponent) %}
-  {% if innerComponent %}
+  {% if innerComponent | trim | length > 0 %}
```
Comment (alibombaye) [inline app/views/preview/components/preview-section-table-row.njk]: oooh interesting. Is this what was need to not show when nothing is passed in?

File: app/views/preview/sections/preview-browser-based.test.js [hunk 3]
```diff
@@ -56,4 +56,104 @@ describe('preview-browser-based', () => {
...
+        const $ = cheerio.load(res.text);
+
+        const browserBasedSectionTable = $('[data-test-id="preview-section-table-browser-based"]');
+        const pluginsRequiredQuestionRow = browserBasedSectionTable.find('[data-test-id="preview-section-table-row-plugins-detail"]');
+
+        expect(browserBasedSectionTable.length).toEqual(1);
+        expect(pluginsRequiredQuestionRow.length).toEqual(1);
+        expect(pluginsRequiredQuestionRow
+          .find('.nhsuk-summary-list__key').text().trim()).toEqual('Plug-ins or extensions information');
+        expect(pluginsRequiredQuestionRow.find('div[data-test-id="preview-question-data-text"]').length).toEqual(1);
```
Comment (alibombaye) [inline app/views/preview/sections/preview-browser-based.test.js]: You don't actually check to if the answer provided in the context is rendered.

File: integration-tests/showBrowsersSupportedPage.test.js [hunk 4]
```diff
@@ -160,6 +161,56 @@ test('should render the validation errors indicating the supported browsers and
...
+
+  await t
+    .expect(errorSummary.exists).notOk()
+    .click(submitButton.find('button'))
+    .expect(errorSummary.exists).ok()
+    .expect(errorSummaryList.find('li:nth-child(1) a').count).eql(1)
+    .expect(errorSummaryList.find('li:nth-child(1) a').getAttribute('href')).eql('#mobile-responsive')
+    .click(errorSummaryList.find('li:nth-child(1) a'))
+    .expect(getLocation()).contains('/S100000-001/section/browsers-supported#mobile-responsive');
+});
```
Comment (alibombaye) [inline integration-tests/showBrowsersSupportedPage.test.js]: 👍

File: integration-tests/showFeaturePage.test.js [hunk 5]
```diff
@@ -159,6 +159,31 @@ test('should show validation for fields exceeding the maxLength', async (t) => {
     .expect(thirdFieldError.find('.nhsuk-error-message').innerText).eql('Error:\nThis feature is over the character limit');
 });
 
+test('should goto anchor when clicking the client application types required summary error link', async (t) => {
```
Comment (alibombaye) [inline integration-tests/showFeaturePage.test.js]: Change the test description here

File: integration-tests/showPreviewPage.test.js [hunk 6]
```diff
@@ -124,5 +133,80 @@ test('when existing marketing data - The client application type section and bro
...
     .expect(mobileResponsiveRow.find('.nhsuk-summary-list__key').innerText).eql('Mobile responsive')
-    .expect(mobileResponsiveRow.find('.nhsuk-summary-list__value').innerText).eql('Yes');
+    .expect(mobileResponsiveRow.find('.nhsuk-summary-list__value').innerText).eql('Yes')
+    .expect(pluginsRequiredRow.find('.nhsuk-summary-list__key').innerText).eql('Plug-ins or extensions required')
+    .expect(pluginsRequiredRow.find('.nhsuk-summary-list__value').innerText).eql('Yes')
+    .expect(pluginsDetailRow.find('.nhsuk-summary-list__key').innerText).eql('Plug-ins or extensions information')
+    .expect(pluginsDetailRow.find('.nhsuk-summary-list__value').innerText).eql('The plugin detail');
+});
+
+const pluginsOrExtensionsMarketingData = {
```
Comment (alibombaye) [inline integration-tests/showPreviewPage.test.js]: Is there a reason this is not in the fixture file?

File: app/forms/sections/plug-ins-or-extensions.json (hunk 1 above)
Comment (sam-roberts-bjss) [reply app/forms/sections/plug-ins-or-extensions.json]: Correct. I'll remove this.

File: app/views/preview/components/preview-section-table-row.njk (hunk 2 above)
Comment (sam-roberts-bjss) [reply app/views/preview/components/preview-section-table-row.njk]: Yeah this was one of the ways. Basically the innerComponent was still generating whitespace even though a question did not have an answer.

After our discussion I will add a unit test to check this.

File: app/views/preview/sections/preview-browser-based.test.js (hunk 3 above)
Comment (sam-roberts-bjss) [reply app/views/preview/sections/preview-browser-based.test.js]: I'll add this.

File: integration-tests/showFeaturePage.test.js (hunk 5 above)
Comment (sam-roberts-bjss) [reply integration-tests/showFeaturePage.test.js]: Good spot. Will fix this.

**Pull request merged**